### PR TITLE
Recommendation

### PR DESCRIPTION
**Pull request by julien4215**
Add recommendation module

### CODE REVIEW

**Review by maximeroucher — COMMENTED** [review]
Looks awesome !!
Can you add some screenshot to give us an idea of the UI 😉

File: lib/recommendation/ui/pages/add_edit_page.dart [hunk 1]
```diff
@@ -90,13 +91,19 @@ class AddEditRecommendationPage extends HookConsumerWidget {
                         final XFile? image =
                             await picker.pickImage(source: ImageSource.gallery);
                         if (image != null) {
-                          if (kIsWeb) {
-                            logo.value = await image.readAsBytes();
-                            logoFile.value = Image.network(image.path);
+                          final size = await image.length();
+                          if (size > maxFileSize) {
+                            displayAdvertToastWithContext(TypeMsg.error,
+                                RecommendationTextConstants.imageSizeTooBig);
```
Comment (maximeroucher) [inline lib/recommendation/ui/pages/add_edit_page.dart:95]: Oh, this is awesome !!
Can you add it to the other modules, or even extract this image picker to the tools ?
I am assigning you an enhancement issue 😉

**Review by maximeroucher — COMMENTED** [review]
Amazing work !!
Your commits are really well done !
Just waiting Hyperion to approve this

**Review by maximeroucher — APPROVED** [review]

**Review by guillaumetavernier — APPROVED** [review]

File: lib/recommendation/ui/widgets/recommendation_card.dart [hunk 2]
```diff
@@ -69,37 +69,43 @@ class RecommendationCard extends HookConsumerWidget {
...
-                            displayToastWithContext(
-                              TypeMsg.msg,
-                              RecommendationTextConstants.copiedCode,
-                            );
-                          },
-                          icon: const Icon(Icons.copy),
-                        ),
-                    ],
-                  ),
+                        if (recommendation.code != null)
```
Comment (maximeroucher) [inline lib/recommendation/ui/widgets/recommendation_card.dart:94]: This condition is a duplicate, you are applying it for the whole Row

**Review by maximeroucher — APPROVED** [review]

File: lib/recommendation/ui/widgets/recommendation_card.dart (hunk 2 above)
Comment (julien4215) [reply lib/recommendation/ui/widgets/recommendation_card.dart:94]: Fixed in #318

**Pull request merged**